syntax = "proto3";                                                                                                                                                                                              
package DAG;


message mut {
    int32 position = 1; // Position in the chromosome
    /* All nucleotides are encoded as integers (0:A, 1:C, 2:G, 3:T) */
    int32 par_nuc = 2; // Nucleotide of parent at this position
    repeated int32 mut_nuc = 3; // Mutated nucleotide in this node at this position

    string chromosome = 5; // Chromosome string. Currently unused.
}

message edge {
    int64 parent_node = 2; 
    int64 parent_clade = 3;
    int64 child_node = 4; 
    repeated mut edge_mutations = 5; 
    float edge_weight = 6; // probability in node-clade pair? other weight?
}

/// Data about each node.
message node_name {
    repeated string condensed_leaves = 2; // A list of strings for the names of the sequence/sequences which are represented by the node above. can be just one if a leaf is unique. 
}

/// removed the newick since the DAG will have nodes that cannot be represented within a single tree anyway.  
message data {
    repeated edge edges = 1; /// since we are explicitly referencing each node and each edge,
    repeated node_name node_names = 2; /// Parent_node and child_node edge data refers to index in this list of nodes
    string reference_id = 3; /// the name for the implied sequence at the UA node
    string reference_seq = 4; /// the reference sequence at the UA node
}
